### PR TITLE
[#5139] optimize remove folder

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a resource. Upload couple of files to the root of the resource. 
2. Create a folder in the same resource. Upload few files to that folder. 
3. Create another folder and upload files to that folder. Make this folder a fileset aggregation.
4. Download the resource bag and verify the resourcemap.xml file correctly describes the files and folders of the resource.
5. Delete the folder that you created in step 2. Verify the folder and the files in that folder got deleted.
6. Download the resource bag and verify the resourcemap.xml file correctly describes the files and folders of the resource.
7. Delete the folder that you created in step 3. Verify the folder and the files in that folder got deleted.
8. Download the resource bag and verify the resourcemap.xml file correctly describes the files and folders of the resource.

### CODE REVIEW

**Review by devincowan — CHANGES_REQUESTED** [review]
One question and one small fix

**Comment by devincowan** [comment]
@pkdash the iRods AUTH failures on this PR build are odd
http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/6546/
have you seen this before?

File: hs_file_types/models/model_instance.py [hunk 1]
```diff
@@ -571,12 +571,17 @@ def copy_executed_by(self, tgt_logical_file):
...
-        super(ModelInstanceLogicalFile, self).logical_delete(user, delete_res_files=delete_res_files)
+        if delete_meta_files:
+            istorage = self.resource.get_irods_storage()
+            if istorage.exists(self.schema_values_file_path):
+                istorage.delete(self.schema_values_file_path)
+        super(ModelInstanceLogicalFile, self).logical_delete(
+            user,
+            delete_res_files=delete_res_files,
+            delete_meta_files=delete_meta_files
+        )
```
[the remark this answers is not in the document]
Comment (pkdash) [reply hs_file_types/models/model_instance.py:584]: Line 577 should have been `istorage = resource.get_irods_storage()`. Will fix that.

File: hs_core/views/utils.py [hunk 2]
```diff
@@ -902,25 +902,48 @@ def remove_irods_folder_in_django(resource, folder_path, user):
...
-        hydroshare.delete_format_metadata_after_delete_file(resource, file_name)
+            if lf.aggregation_name == rel_folder_path:
+                lf.logical_delete(user, resource=resource, delete_res_files=False, delete_meta_files=False)
+                break
+
+    deleted_file_extensions = {get_file_extension(f) for f in res_file_set}
+
+    # delete resource file records from Django DB
+    ResourceFile.objects.filter(object_id=resource.id,
+                                file_folder__startswith=rel_folder_path).delete()
```
[the remark this answers is not in the document]
Comment (pkdash) [reply hs_core/views/utils.py:935]: Deleting resource files as part of deleting the aggregation/logical file will result in additional queries. Even then the files that are not part of any aggregation still need to be deleted in a separate query. So it's better to delete all files in one query.

[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/views/utils.py:935]: makes sense thanks

File: hs_file_types/models/model_instance.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_file_types/models/model_instance.py]: Yep, but do you also want to pass `resource` as a parameter into `super(ModelInstanceLogicalFile, self).logical_delete(...)`?

**Review by devincowan — DISMISSED** [review]

File: hs_file_types/models/model_instance.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkdash) [reply hs_file_types/models/model_instance.py]: For sure. Fixed it now finally.

**Review by devincowan — APPROVED** [review]
this is great, thanks pabitra!

**Pull request merged**